hudi-common/src/main/java/org/apache/hudi/common/util/queue/DisruptorBasedProducer.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.common.util.queue;
+
+public abstract class DisruptorBasedProducer<I> implements HoodieProducer<I> {
```

Comment:
We should make both producers/consumers invariant of the Queue impl, in other words we should keep implementation details (BoundedLockingQueue, Disruptor, etc) abstracted away from them (there're simply no reason for them to be aware what's the implementation details of the queue).

 - We should keep only `HoodieQueueProducer` (removing specific ones)
 - Same for Consumers

Reply:
Nice catch! totally agree!

abstract HoodieConsumer and HoodieProducer. They modify HoodieMessageQueue and don't care what the specific implementation of HoodieMessageQueue is